novawallet/Modules/Transfer/BaseTransfer/OnChain/OnChainTransferInteractor.swift

```diff
@@ -535,6 +547,26 @@ extension OnChainTransferInteractor {
         }
     }
 
+    func requestFeePaymentAvailability(for chainAsset: ChainAsset) {
+        let wrapper = transferAggregationWrapperFactory.createCanPayFeeWrapper(in: chainAsset)
+
+        execute(
+            wrapper: wrapper,
+            inOperationQueue: operationQueue,
+            runningCallbackIn: .main,
+            callbackClosure: { [weak self] result in
+                switch result {
+                case let .success(available):
+                    self?.presenter?.didReceiveSendingAssetFeeAvailable(available)
+                case let .failure(error) where error is AssetConversionAggregationFactoryError:
+                    self?.presenter?.didReceiveSendingAssetFeeAvailable(false)
```

Comment:
The wrapper factory encapsulates the logic for checking whether `hasAssetHubTransferFees` or `hasHydrationTransferFees` is available. If neither is available, it wraps the error. I believe this approach is better since it's a common logic, so we don't need to implement it at the upper level, such as in interactors.